src/state.rs

```diff
@@ -25,60 +28,33 @@ impl<const DEPTH: usize> Default for State<DEPTH> {
 impl<const DEPTH: usize> State<DEPTH> {
     pub fn new() -> State<DEPTH> {
         State {
-            tree: PoseidonTree::<DataLeaf, (), DEPTH>::default(),
+            tree: Tree::<Unit, DEPTH, ARITY>::new(),
+            licenses: BTreeMap::new(),
         }
     }
     pub fn append_license(&mut self, lic: &License) {
-        self.tree.push(DataLeaf { lic: lic.clone() });
-    }
+        let lpk = JubJubAffine::from(lic.lsa.pk_r().as_ref());
 
-    // TODO: we should update this function when starting to use the new
-    // implementation of the Merkle tree
-    pub fn get_licenses(&self, vk: &ViewKey) -> Vec<License> {
-        let mut lic_vec = Vec::new();
-        for i in 0u64..(4u64.pow(DEPTH as u32)) {
-            let leaf = self.tree.get(i);
-
-            match leaf {
-                Some(leaf) if vk.owns(&leaf.lic.lsa) => {
-                    lic_vec.push(leaf.lic);
-                }
-                _ => break,
-            }
-        }
+        let item = PoseidonItem {
+            hash: sponge::hash(&[lpk.get_x(), lpk.get_y()]),
+            data: Unit,
+        };
 
-        lic_vec
+        self.tree.insert(0, item);
+        self.licenses.insert(lic.pos, lic.clone());
```

Comment:
before this line, we need to set the position of the license, for instance, as suggested above, `lic.pos = licenses.len()`